vision_agent/tools/tools.py

```diff
@@ -1878,6 +1878,74 @@ def closest_box_distance(
+    """'document_analysis' is a document understanding tool that can handle and analyse various
+    types of document images with text, tables, pictures, charts and other information.
+    It returns a structured output containing the text, tables, pictures, charts and information
+    caption, summary, labels, bounding boxes avoiding information loss.
+
+    Parameters:
+        image (np.ndarray): The document image to analyze
+
+    Returns:
+        Dict[str, Any]: A dictionary containing the extracted text, tables, and other
+            information.
+
+    Example
+    -------
+        >>> document_analysis(image)
+        {'filename': None,
+         'pages': [{'bbox': [left, top, right, bottom],
+                    'chunks': [{'bbox': [left, top, right, bottom],
+                                'caption': 'TITLE',
+                                'label': 'page_header',
+                                'summary': 'The image contains a single word:\n'
+                                        '\n'
+                                        'TITLE\n'
+                                        '\n'
+                                        'The logo or brand name, '
+                                        'presented in a simple, bold, all-capital '
+                                        'letter font against a plain background.'},
+                                {'bbox': [left, top, right, bottom],
+                                 'caption': {'annotation': 'Cumulative Cloud CapEx Spending',
+                                    'data': [{'value': 200, 'year': '2024'},
+                                             {'value': 250, 'year': '2025'}],
+                                    'notes': 'Values for 2024 and 2025 are '
+                                             "estimates, indicated by 'E'",
+                                    'title': 'Total CapEx Spending',
+                                    'trend': 'Generally increasing over time',
+                                    'type': 'bar chart',
+                                    'unit': 'Billion USD',
+                                    'xAxis': 'Year',
+                                    'yAxis': 'Total CapEx Spending'},
+                                'label': 'picture',
+                        '       summary': 'This bar chart illustrates the trend of '
+                                   'Total CapEx (Capital Expenditure) Spending '
+                                   'over time, from 2024 to 2025.'},
+
+                    (...)
+
+    """
+
+    image_file = numpy_to_bytes(image)
+
+    files = [("image", image_file)]
+
+    payload = {
+        "model": "document-analysis",
+    }
+
+    response: dict[str, Any] = send_inference_request(
+        payload=payload,
+        endpoint_name="document-analysis",
+        files=files,
+        v2=True,
+        metadata_payload={"function_name": "document_analysis"},
+    )
+
```

Comment:
One thing that just got added is this this function https://github.com/landing-ai/vision-agent/blob/main/vision_agent/tools/tools.py#L203

This is to re-display the data for the front end to use it. Could you add this as well?

Reply:
Ok, added! Thanks